core/drivers/crypto/se050/adaptors/utils/utils.c

```diff
@@ -0,0 +1,299 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) Foundries Ltd. 2020 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+
+#include <crypto/crypto.h>
+#include <se050.h>
+#include <string.h>
+
+#define BIT(nr) (1UL << (nr))
+
+/* base value for secure objects (transient and persistent) */
+#define OID_MIN			((uint32_t)(0x00000001))
+#define OID_MAX			((uint32_t)(OID_MIN + 0x7BFFFFFE))
+#define NBR_OID			((uint32_t)(OID_MAX - OID_MIN))
+
+#define IS_WATERMARKED(x)	(((x) & WATERMARKED(0)) == WATERMARKED(0))
+
+static void delete_transient_objects(void)
+{
+	Se05xSession_t *ctx = NULL;
+	uint8_t more = kSE05x_MoreIndicator_NA;
+	uint8_t *list = NULL;
+	size_t len = 1024;
+	smStatus_t status  = SM_NOT_OK;
+	uint16_t offset = 0;
+	uint32_t id = 0;
+	size_t nbr_pobj = 0, nbr_tobj = 0;
+	size_t i = 0;
+
+	if (!se050_session)
+		return;
+
+	ctx = &se050_session->s_ctx;
+
+	list = calloc(1, len);
+	if (!list)
+		return;
+	do {
+		status = Se05x_API_ReadIDList(ctx, offset, 0xFF, &more,
+					      list, &len);
+		if (status != SM_OK) {
+			free(list);
+			return;
+		}
+
+		offset = len;
+		for (i = 0; i < len; i += 4) {
```

Comment:
check that `len` is a multiple of 4 with a minimum value of 4.

Reply:
this is the recommended way of processing the response to the API....I think it will be redundant

Reply:
added check anyway